### ISSUE TITLE
The current implementation containing bugs on dataloader.

### ISSUE THREAD

**Issue by BestSonny** [facebookresearch/DepthContrast#14]
Dear authors,

I think the repo is an excellent implementation of your excellent paper. However, it shares a common bug in setting the seed number for data augmentation as pointed in [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/6960) and [here](https://github.com/pytorch/pytorch/pull/56488)

I have tested in this repo and found out that each sample will only have two one-time augmented versions across the whole training epochs, which is problematic to the self-contrastive learning setting.

You may want to revisit the experiments in the paper.

Best,
Pan He

**Comment by zaiweizhang** [comment]
Hi Pan,

Thanks for pointing that out! That bug doesn’t occur if we use the spawn start method like the following:

```
from torch.multiprocessing import Pool, Process, set_start_method
try:
     set_start_method('spawn')
except RuntimeError:
    pass
```

For all experiments conducted in the paper, we use the spawn setting. We will fix the bug in the released code soon.

Thanks,
Zaiwei

**Comment by BestSonny** [comment]
@zaiweizhang  Great. Looking forward to your updated code and results. Thanks

**Comment by baraujo98** [comment]
So calling main with `--multiprocessing-distributed` solves the problem in this case, correct?

**Comment by zaiweizhang** [comment]
Actually, you also need to change the setting to spawn. We have changed it in the main.py. Please try again! Sorry for the trouble.

**Comment by baraujo98** [comment]
Ok thanks for the information. Does this bug occur across all pytorch frameworks? For example, does this happen when finetuning with OpenPCDet?

**Comment by BestSonny** [comment]
@baraujo98  As mentioned, this is a common bug that impacts a lot of PyTorch frameworks. A blog has discussed this as well [here](https://tanelp.github.io/posts/a-bug-that-plagues-thousands-of-open-source-ml-projects/). 

"The bug is easy to make. In some cases, it has minimal effect on final performance. In others, the identical augmentations can cause severe degradations."

**Comment by zaiweizhang** [comment]
@baraujo98 I do not think that it will affect the fine-tuning with OpenPCDet. It's better to check their [codebase](https://github.com/open-mmlab/OpenPCDet).

**Comment by baraujo98** [comment]
Ok, it _looks_ to me like OpenPCDet does not set a fixed seed by default, hinted by the fact that a `--fix_random_seed` argument exists, and it is False by default. Is that enough to be safe or should I make sure the seed is being reset after each epoch? Thanks for the heads up @BestSonny and for the fast correction @zaiweizhang

**Comment by zaiweizhang** [comment]
Feel free to reopen. Closing it for now.
